src/ng/q.js

```diff
@@ -254,25 +254,34 @@ function qFactory(nextTick, exceptionHandler) {
    *
    * @returns {Deferred} Returns a new instance of deferred.
    */
-  var defer = function() {
-    var d = new Deferred();
+  var defer = function(keepTrack) {
+    var d = new Deferred(keepTrack);
     //Necessary to support unbound execution :/
     d.resolve = simpleBind(d, d.resolve);
     d.reject = simpleBind(d, d.reject);
     d.notify = simpleBind(d, d.notify);
     return d;
   };
 
-  function Promise() {
+  function Promise(keepTrack) {
```

Comment:
Why this flag? this is enabled by default and is only tracking the count so I don't see any reason for having this flag unnecessarily. Let's remove this

Reply:
Why don't you want to track even those promises? That seems like a very specific requirement that doesn't apply in general. You may not care even if for those scenarios promises are leaking but not tracking that isn't the correct solution instead if you really want to get that number then that should really be tracked as a separate number and then you can remove those instances from total pending count in your calculations in telemetry

Reply:
That makes sense. I think that requirement was more geared towards actually tracking the promises in memory and not counting them. I will revert those changes as well. Thanks for the feedback 😄